File: Array/liuchang/day46/day46.java
public class day46 {

    public static void moveZeroes(int[] nums) {
        int lastNumber = 0;
        for (int i = 0; i < nums.length; ++i) {
            if (nums[i] != 0) {
                nums[lastNumber] = nums[i];
                if (i != lastNumber) {
                    nums[i] = 0;
                }
                lastNumber++;
            }


        }
        for (int j = 0; j < nums.length; ++j) {
            System.out.println("=====" + nums[j]);
        }
    }

    public static void main(String[] args) {
        moveZeroes(new int[]{1,0,3,0,10});
    }
}

File: Array/liuchang/day46/day46.kt
package net.study.kotlintest

/**
 * @author       刘畅
 * @createdate   2020/7/16
 * @describe
 */
class day46 {

    private fun moveZeroes(nums:IntArray){
        var lastNumber = 0;
        for(i in nums.indices){
            if (nums[i] != 0) {
                nums[lastNumber] = nums[i];
                if (i != lastNumber) {
                    nums[i] = 0;
                }
                lastNumber++;
            }


        }
        for (element in nums){
            println("=====$element");
        }
    }

    fun main(args:Array<String>){
        moveZeroes(intArrayOf(1,0,3,0,10));
    }
}